@frenck reported CI failures since the merge of the netatmo select PR.

```
hass = <homeassistant.core.HomeAssistant object at 0x7ff3845eb9a0>
config_entry = <tests.common.MockConfigEntry object at 0x7ff39aaff4c0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x7ff384b39ac0>
netatmo_auth = None

    async def test_select_schedule_thermostats(hass, config_entry, caplog, netatmo_auth):
        """Test service for selecting Netatmo schedule with thermostats."""
        with selected_platforms(["climate", "select"]):
            await hass.config_entries.async_setup(config_entry.entry_id)
    
            await hass.async_block_till_done()
    
        webhook_id = config_entry.data[CONF_WEBHOOK_ID]
        select_entity = "select.netatmo_myhome"
    
>       assert hass.states.get(select_entity).state == "Default"
E       AttributeError: 'NoneType' object has no attribute 'state'
```

--- a/homeassistant/components/netatmo/climate.py
+++ b/homeassistant/components/netatmo/climate.py
@@ -261,7 +261,8 @@ class NetatmoThermostat(NetatmoBase, ClimateEntity):
         assert device
         self.hass.data[DOMAIN][DATA_DEVICE_IDS][self._home_id] = device.id
 
-    async def handle_event(self, event: dict) -> None:
+    @callback
+    def handle_event(self, event: dict) -> None:
         """Handle webhook events."""
         data = event["data"]
 

--- a/homeassistant/components/netatmo/climate.py
+++ b/homeassistant/components/netatmo/climate.py
@@ -58,6 +58,7 @@ from .data_handler import (
     HOMESTATUS_DATA_CLASS_NAME,
     NetatmoDataHandler,
 )
+from .helper import get_all_home_ids
 from .netatmo_entity_base import NetatmoBase
 
 _LOGGER = logging.getLogger(__name__)
@@ -131,9 +132,6 @@ async def async_setup_entry(
     if not home_data or home_data.raw_data == {}:
         raise PlatformNotReady
 
-    if HOMEDATA_DATA_CLASS_NAME not in data_handler.data:
-        raise PlatformNotReady
-
     entities = []
     for home_id in get_all_home_ids(home_data):
         for room_id in home_data.rooms[home_id]:
@@ -145,12 +143,18 @@ async def async_setup_entry(
             if home_status and room_id in home_status.rooms:
                 entities.append(NetatmoThermostat(data_handler, home_id, room_id))
 
-        hass.data[DOMAIN][DATA_SCHEDULES][home_id] = {
-            schedule_id: schedule_data.get("name")
-            for schedule_id, schedule_data in (
-                data_handler.data[HOMEDATA_DATA_CLASS_NAME].schedules[home_id].items()
-            )
-        }
+        hass.data[DOMAIN][DATA_SCHEDULES].update(
+            {
+                home_id: {
+                    schedule_id: schedule_data.get("name")
+                    for schedule_id, schedule_data in (
+                        data_handler.data[HOMEDATA_DATA_CLASS_NAME]
+                        .schedules[home_id]
+                        .items()
+                    )
+                }
+            }
+        )
 
     hass.data[DOMAIN][DATA_HOMES] = {
         home_id: home_data.get("name")
@@ -617,14 +621,3 @@ class NetatmoThermostat(NetatmoBase, ClimateEntity):
         device_info: DeviceInfo = super().device_info
         device_info["suggested_area"] = self._room_data["name"]
         return device_info
-
-
-def get_all_home_ids(home_data: pyatmo.HomeData | None) -> list[str]:
-    """Get all the home ids returned by NetAtmo API."""
-    if home_data is None:
-        return []
-    return [
-        home_data.homes[home_id]["id"]
-        for home_id in home_data.homes
-        if "modules" in home_data.homes[home_id]
-    ]

--- a/homeassistant/components/netatmo/climate.py
+++ b/homeassistant/components/netatmo/climate.py
@@ -58,7 +58,7 @@ from .data_handler import (
     HOMESTATUS_DATA_CLASS_NAME,
     NetatmoDataHandler,
 )
-from .helper import get_all_home_ids
+from .helper import get_all_home_ids, update_climate_schedules
 from .netatmo_entity_base import NetatmoBase
 
 _LOGGER = logging.getLogger(__name__)
@@ -143,18 +143,12 @@ async def async_setup_entry(
             if home_status and room_id in home_status.rooms:
                 entities.append(NetatmoThermostat(data_handler, home_id, room_id))
 
-        hass.data[DOMAIN][DATA_SCHEDULES].update(
-            {
-                home_id: {
-                    schedule_id: schedule_data.get("name")
-                    for schedule_id, schedule_data in (
-                        data_handler.data[HOMEDATA_DATA_CLASS_NAME]
-                        .schedules[home_id]
-                        .items()
-                    )
-                }
-            }
+    hass.data[DOMAIN][DATA_SCHEDULES].update(
+        update_climate_schedules(
+            home_ids=get_all_home_ids(home_data),
+            schedules=data_handler.data[HOMEDATA_DATA_CLASS_NAME].schedules,
         )
+    )
 
     hass.data[DOMAIN][DATA_HOMES] = {
         home_id: home_data.get("name")

--- a/homeassistant/components/netatmo/helper.py
+++ b/homeassistant/components/netatmo/helper.py
@@ -1,6 +1,10 @@
 """Helper for Netatmo integration."""
+from __future__ import annotations
+
 from dataclasses import dataclass
 from uuid import UUID, uuid4
+
+import pyatmo
 
 
 @dataclass
@@ -15,3 +19,14 @@ class NetatmoArea:
     mode: str
     show_on_map: bool
     uuid: UUID = uuid4()
+
+
+def get_all_home_ids(home_data: pyatmo.HomeData | None) -> list[str]:
+    """Get all the home ids returned by NetAtmo API."""
+    if home_data is None:
+        return []
+    return [
+        home_data.homes[home_id]["id"]
+        for home_id in home_data.homes
+        if "modules" in home_data.homes[home_id]
+    ]

--- a/homeassistant/components/netatmo/helper.py
+++ b/homeassistant/components/netatmo/helper.py
@@ -30,3 +30,14 @@ def get_all_home_ids(home_data: pyatmo.HomeData | None) -> list[str]:
         for home_id in home_data.homes
         if "modules" in home_data.homes[home_id]
     ]
+
+
+def update_climate_schedules(home_ids: list[str], schedules: dict) -> dict:
+    """Get updated list of all climate schedules."""
+    return {
+        home_id: {
+            schedule_id: schedule_data.get("name")
+            for schedule_id, schedule_data in schedules[home_id].items()
+        }
+        for home_id in home_ids
+    }

--- a/homeassistant/components/netatmo/select.py
+++ b/homeassistant/components/netatmo/select.py
@@ -22,7 +22,7 @@ from .const import (
     SIGNAL_NAME,
 )
 from .data_handler import HOMEDATA_DATA_CLASS_NAME, NetatmoDataHandler
-from .helper import get_all_home_ids
+from .helper import get_all_home_ids, update_climate_schedules
 from .netatmo_entity_base import NetatmoBase
 
 _LOGGER = logging.getLogger(__name__)
@@ -42,20 +42,12 @@ async def async_setup_entry(
     if not home_data or home_data.raw_data == {}:
         raise PlatformNotReady
 
-    for home_id in get_all_home_ids(home_data):
-        hass.data[DOMAIN][DATA_SCHEDULES].update(
-            {
-                home_id: {
-                    schedule_id: schedule_data.get("name")
-                    for schedule_id, schedule_data in (
-                        data_handler.data[HOMEDATA_DATA_CLASS_NAME]
-                        .schedules[home_id]
-                        .items()
-                    )
-                }
-            }
+    hass.data[DOMAIN][DATA_SCHEDULES].update(
+        update_climate_schedules(
+            home_ids=get_all_home_ids(home_data),
+            schedules=data_handler.data[HOMEDATA_DATA_CLASS_NAME].schedules,
         )
-    _LOGGER.debug("Adding climate schedules %s", hass.data[DOMAIN][DATA_SCHEDULES])
+    )
 
     entities = [
         NetatmoScheduleSelect(

--- a/homeassistant/components/netatmo/select.py
+++ b/homeassistant/components/netatmo/select.py
@@ -13,7 +13,6 @@ from homeassistant.exceptions import PlatformNotReady
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .climate import get_all_home_ids
 from .const import (
     DATA_HANDLER,
     DATA_SCHEDULES,
@@ -23,6 +22,7 @@ from .const import (
     SIGNAL_NAME,
 )
 from .data_handler import HOMEDATA_DATA_CLASS_NAME, NetatmoDataHandler
+from .helper import get_all_home_ids
 from .netatmo_entity_base import NetatmoBase
 
 _LOGGER = logging.getLogger(__name__)
@@ -42,8 +42,20 @@ async def async_setup_entry(
     if not home_data or home_data.raw_data == {}:
         raise PlatformNotReady
 
-    if HOMEDATA_DATA_CLASS_NAME not in data_handler.data:
-        raise PlatformNotReady
+    for home_id in get_all_home_ids(home_data):
+        hass.data[DOMAIN][DATA_SCHEDULES].update(
+            {
+                home_id: {
+                    schedule_id: schedule_data.get("name")
+                    for schedule_id, schedule_data in (
+                        data_handler.data[HOMEDATA_DATA_CLASS_NAME]
+                        .schedules[home_id]
+                        .items()
+                    )
+                }
+            }
+        )
+    _LOGGER.debug("Adding climate schedules %s", hass.data[DOMAIN][DATA_SCHEDULES])
 
     entities = [
         NetatmoScheduleSelect(
@@ -51,8 +63,7 @@ async def async_setup_entry(
             home_id,
             list(hass.data[DOMAIN][DATA_SCHEDULES][home_id].values()),
         )
-        for home_id in get_all_home_ids(home_data)
-        if home_id in hass.data[DOMAIN][DATA_SCHEDULES]
+        for home_id in hass.data[DOMAIN][DATA_SCHEDULES]
     ]
 
     _LOGGER.debug("Adding climate schedule select entities %s", entities)

--- a/homeassistant/components/netatmo/select.py
+++ b/homeassistant/components/netatmo/select.py
@@ -116,7 +116,8 @@ class NetatmoScheduleSelect(NetatmoBase, SelectEntity):
                 )
             )
 
-    async def handle_event(self, event: dict) -> None:
+    @callback
+    def handle_event(self, event: dict) -> None:
         """Handle webhook events."""
         data = event["data"]
 

--- a/tests/components/netatmo/test_select.py
+++ b/tests/components/netatmo/test_select.py
@@ -18,6 +18,8 @@ async def test_select_schedule_thermostats(hass, config_entry, caplog, netatmo_a
     webhook_id = config_entry.data[CONF_WEBHOOK_ID]
     select_entity = "select.netatmo_myhome"
 
+    assert hass.states.get(select_entity).state == "Default"
+
     # Fake backend response changing schedule
     response = {
         "event_type": "schedule",
@@ -26,6 +28,7 @@ async def test_select_schedule_thermostats(hass, config_entry, caplog, netatmo_a
         "push_type": "home_event_changed",
     }
     await simulate_webhook(hass, webhook_id, response)
+    await hass.async_block_till_done()
 
     assert hass.states.get(select_entity).state == "Winter"
     assert hass.states.get(select_entity).attributes[ATTR_OPTIONS] == [

